### PR TITLE
Moving from DPC++ compiler to ICX/ICPX usage

### PR DESCRIPTION
**Pull request by akabalov**
We started to get deprecation's warning that we need to move from usage of DPC++ compiler to ICPX on Linux and ICX on Windows. 

This PR is addressing this request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,9 +89,11 @@ oneapi::mkl::blas::column_major::gemm(gpu_queue, transA, transB, m, ...);
 ```
 How to build an application with run-time dispatching:
 
+if OS is Linux, use icpx compiler. If OS is Windows, use icx compiler.
+Windows example:
 ```cmd
-$> dpcpp -fsycl –I$ONEMKL/include app.cpp
-$> dpcpp -fsycl app.o –L$ONEMKL/lib –lonemkl
+$> icx -fsycl –I$ONEMKL/include app.cpp
+$> icx -fsycl app.o –L$ONEMKL/lib –lonemkl
```
Comment (mkrainiuk) [inline README.md]: This is linux example, could you please use icpx?

**Review by mkrainiuk — APPROVED** [review]
Thanks for PR! Looks good to me with one minor change in README example.

File: README.md (hunk 1 above)
Comment (akabalov) [reply README.md]: Good catch! Thank you! Changed to icpx

File: CMakeLists.txt [hunk 2]
```diff
@@ -93,18 +93,31 @@ else()
...
+        message(STATUS "CXX compiler: icpx was found in PATH, using icpx")
+        set(CMAKE_CXX_COMPILER "icpx")
+      else()
+        message(STATUS "CXX compiler: icx was found in PATH, using icx")
+        set(CMAKE_CXX_COMPILER "icx")
+      endif()
     else()
       if(WIN32)
-        message(STATUS "CXX compiler: dpcpp was not found in PATH, using clang-cl instead")
+        message(STATUS "CXX compiler: icpx/icx was not found in PATH, using clang-cl instead")
```
Comment (dnhsieh-intel) [inline CMakeLists.txt]: I think we could put `icx` here to be more specific.

File: CMakeLists.txt [hunk 3]
```diff
@@ -93,18 +93,31 @@ else()
...
         set(CMAKE_CXX_COMPILER "clang-cl")
       else()
-        message(STATUS "CXX compiler: dpcpp was not found in PATH, using clang++ instead")
+        message(STATUS "CXX compiler: icpx/icx was not found in PATH, using clang++ instead")
```
Comment (dnhsieh-intel) [inline CMakeLists.txt]: `icpx`

File: deps/googletest/cmake/internal_utils.cmake [hunk 4]
```diff
@@ -83,7 +83,7 @@ macro(config_compiler_and_linker)
     set(cxx_base_flags "-Wall -Wshadow -Wconversion")
     set(cxx_exception_flags "-fexceptions")
     set(cxx_no_exception_flags "-fno-exceptions")
-    set(cxx_strict_flags "-W -Wpointer-arith -Wreturn-type -Wcast-qual -Wwrite-strings -Wswitch -Wunused-parameter -Wcast-align -Wchar-subscripts -Winline -Wredundant-decls")
+    set(cxx_strict_flags "-W -Wpointer-arith -Wreturn-type -Wcast-qual -Wwrite-strings -Wswitch -Wunused-parameter -Wcast-align -Winline -Wredundant-decls")
```
Comment (dnhsieh-intel) [inline deps/googletest/cmake/internal_utils.cmake:86]: What happened to `-Wchar-subscripts`?

**Review by dnhsieh-intel — APPROVED** [review]
Thank you for the PR! Please see the following minor comments. We may also need to change the documentation [Selecting a Compiler](https://github.com/oneapi-src/oneMKL/blob/develop/docs/selecting_a_compiler.rst).

File: CMakeLists.txt (hunk 2 above)
Comment (akabalov) [reply CMakeLists.txt]: Thank you! Changed it to icx

File: CMakeLists.txt (hunk 3 above)
Comment (akabalov) [reply CMakeLists.txt]: Thank you! Changed it to icpx

File: deps/googletest/cmake/internal_utils.cmake (hunk 4 above)
Comment (akabalov) [reply deps/googletest/cmake/internal_utils.cmake:86]: This option is deprecated

**Comment by akabalov** [comment]
> Thank you for the PR! Please see the following minor comments. We may also need to change the documentation [Selecting a Compiler](https://github.com/oneapi-src/oneMKL/blob/develop/docs/selecting_a_compiler.rst).

Thank you a lot! I changed suggested file as well. Let me know if it looks good for you.

File: docs/selecting_a_compiler.rst [hunk 5]
```diff
@@ -7,11 +7,11 @@ You must choose a compiler according to the required backend of your
 application.
 
 * If your application requires Intel GPU, use
-  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_ ``dpcpp``.
+  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_ ``icpx on Linux or icx on Windows``.
```
Comment (dnhsieh-intel) [inline docs/selecting_a_compiler.rst]: Let me be picky here.
```suggestion
  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_ ``icpx`` on Linux or ``icx`` on Windows.
```

File: docs/selecting_a_compiler.rst [hunk 6]
```diff
@@ -7,11 +7,11 @@ You must choose a compiler according to the required backend of your
...
 * If your application requires NVIDIA GPU, use the latest release of
   ``clang++`` from `Intel project for LLVM* technology <https://github.com/intel/llvm/releases>`_.
 * If your application requires AMD GPU, use ``hipSYCL`` from the `hipSYCL repository <https://github.com/illuhad/hipSYCL>`_
 * If no Intel GPU, NVIDIA GPU, or AMD GPU is required, you can use either
   `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_
-  ``dpcpp``, ``clang++``, or ``hipSYCL`` on Linux and ``clang-cl`` on Windows from
+  ``icpx on Linux or icx on Windows``, ``clang++``, or ``hipSYCL`` on Linux and ``clang-cl`` on Windows from
   `Intel project for LLVM* technology <https://github.com/intel/llvm/releases>`_.
```
Comment (dnhsieh-intel) [inline docs/selecting_a_compiler.rst]: This becomes tricky... I would suggest
```
* If no Intel GPU, NVIDIA GPU, or AMD GPU is required, on Linux you can use either
  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_
  ``icpx``, ``clang++``, or ``hipSYCL``; on Windows you can use either
  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_
  ``icx`` or ``clang-cl`` from
  `Intel project for LLVM* technology <https://github.com/intel/llvm/releases>`_.
```

**Review by dnhsieh-intel — APPROVED** [review]
More annoying comments. Looks good to me other than that. Thank you!

File: docs/selecting_a_compiler.rst (hunk 6 above)
Comment (akabalov) [reply docs/selecting_a_compiler.rst]: Done! Thank you a lot, Dai-Ni!

**Pull request merged**